src/core/reference/include/openvino/reference/inverse.hpp

```diff
@@ -0,0 +1,160 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <cstdlib>
+#include <openvino/core/shape.hpp>
+#include <vector>
+
+namespace ov {
+namespace reference {
+namespace inverse {
+
+template <typename T>
+void lu_decomposition(const T* input,
+                      std::vector<std::vector<T>>& L,
+                      std::vector<std::vector<T>>& U,
+                      std::vector<T>& P,
+                      bool& sign,
+                      size_t b,
+                      size_t n) {
+    // Make L identity, U a copy of input and P a range(0, n)
+    size_t batch_idx = b * n * n;
+    for (size_t i = 0; i < n; ++i) {
+        P[i] = static_cast<T>(i);
+        L[i][i] = static_cast<T>(1);
+
+        size_t i_idx = i * n;
+        for (size_t j = 0; j < n; ++j) {
+            U[i][j] = input[batch_idx + i_idx + j];
+        }
+    }
+
+    for (size_t k = 0; k < n; ++k) {
+        // Partial Pivoting
+        size_t pivot_row = k;
+        for (size_t i = k + 1; i < n; ++i) {
+            if (std::abs(U[i][k]) > std::abs(U[pivot_row][k])) {
+                pivot_row = i;
+            }
+        }
+
+        if (pivot_row != k) {
+            // Swap rows in L, U (A) and P
+            std::swap(U[k], U[pivot_row]);
+            std::swap(L[k], L[pivot_row]);
+            std::swap(P[k], P[pivot_row]);
+            sign = !sign;
+        }
+
+        for (size_t i = k + 1; i < n; ++i) {
+            L[i][k] = U[i][k] / U[k][k];
+            for (size_t j = k; j < n; ++j) {
+                U[i][j] -= L[i][k] * U[k][j];
+            }
+        }
+    }
+}
+
+template <typename T>
+void lu_solve(T* output,
+              std::vector<std::vector<T>>& L,
+              std::vector<std::vector<T>>& U,
+              std::vector<T>& P,
+              size_t b,
+              size_t n,
+              size_t column) {
+    std::vector<T> B(n, static_cast<T>(0));
+    std::vector<T> X(n, static_cast<T>(0));
+    std::vector<T> Y(n, static_cast<T>(0));
+    B[column] = static_cast<T>(1);
```

Comment:
Shorter and legit in this context.
```suggestion
    std::vector<T> B(n, T{0});
    std::vector<T> X(n, T{0});
    std::vector<T> Y(n, T{0});
    B[column] = T{1};
```

Reply:
Great idea, applied!